azure-quantum/azure/quantum/target/solvers.py

```diff
@@ -238,13 +242,27 @@ def check_valid_problem(self, problem):
                     f"with problem type {problem.problem_type};"
                     f"Try PopulationAnnealing or SubstochasticMonteCarlo."
                 )
+        if problem.serialization_type == "application/x-protobuf":
+            if not self.supports_protobuf() and self.name not in proto_valid_solver_names:
+                raise ValueError(
+                    f"Solver type is not compatible"
+                    f"for serialization with protobuf;"
+                    f"Try PopulationAnnealing or SubstochasticMonteCarlo."
```

Comment:
Is good to point the user to this, however the problem is that it might change and we will forget to update it :(

Reply:
I'll add a running task for myself to track these :)